nixos/modules/tasks/network-interfaces-scripted.nix

```diff
@@ -183,14 +183,15 @@ let
                 ${flip concatMapStrings ips (ip:
                   let
                     cidr = "${ip.address}/${toString ip.prefixLength}";
+                    brd = if ip ? broadcast then (if ip.broadcast == null then "brd +" else "brd ${ip.broadcast}") else "";
```

Comment:
As you set a default value you can be sure that the `broadcast` option exists. You can remove the outer `if` here.

Reply:
IPv6 doesn't have a assignable per-network broadcast address, so the option only exists in IPv4 address blocks.

It would be clearer to have a way to distinguish an `ip` that came from `ipv4.addresses` from the ones that came from `ipv6.addresses`, but as far as I can tell the only way to check that would be to test the type of `ip.prefixLength` or value of `ip.address` and that's more complicated and less clear.